src/views/ApartmentApp/components/StudentApplication/index.js

```diff
@@ -894,14 +823,14 @@ const StudentApplication = ({ userProfile, authentication }) => {
                 submitDialogOpen={submitDialogOpen}
                 submitStatus={applicationDetails.DateSubmitted ? 'success' : submitStatus}
                 unsavedChanges={unsavedChanges}
-                onCloseDialog={handleCloseDialog}
-                onCloseOkay={handleCloseOkay}
-                onDeleteAppAccepted={handleDeleteAppAccepted}
+                onCloseDialog={(_event, reason) => reason !== 'clickaway' && handleCloseDialog()}
+                onCloseOkay={handleCloseDialog}
+                onDeleteAppAccepted={() => deleteApartmentApplication()}
                 onDeleteButtonClick={() => setDeleteDialogOpen(true)}
-                onSaveButtonClick={handleSaveButtonClick}
-                onShowApplication={handleShowApplication}
-                onSubmitAppAccepted={handleSubmitAppAccepted}
-                onSubmitButtonClick={handleSubmitButtonClick}
+                onSaveButtonClick={() => saveApartmentApplication(applicationDetails)}
+                onShowApplication={() => setApplicationCardsOpen(true)}
+                onSubmitAppAccepted={() => submitApplication()}
```

Comment:
For example, here you're already defining an anonymous function. Realistically this could simply be:
```suggestion
                onSubmitAppAccepted={submitApplication}
```

However, I think it would be much cleaner to have the dialog behavior attached to the dialog itself, rather than the data handling callbacks it calls:
```suggestion
                onSubmitAppAccepted={() => {
                    handleCloseDialog();
                    submitApplication();
                }
```

Reply:
Since the instance of the `GordonDialogBox` within the `BottomBar` are do nothing other than directly apply the props passed to the `BottomBar`, I was just thinking it would probably be better to just put those dialog boxes directly in the `StudentApplication`, so because we are just passing all these callbacks through the a subcomponent for no real reason.

I think it would remove a lot of confusion and clutter from the `BottomBar` component